tests/Feature/HealthySqlDumpTest.php

```diff
@@ -65,4 +68,19 @@ public function it_fails_when_sql_process_exceed_the_timeout_limit()
         // Manually remove database, because error exited code before deleting it
         DB::select('DROP DATABASE `healthy-sql-dump--backup--mysite-2019-09-16-08-00-07`');
     }
+
+    /** @test */
+    public function it_fails_because_backup_size_exceeds_disk_space_size()
+    {
+        config()->set('backup.monitor_backups.0.name', 'mysite');
+
+        $statuses = BackupDestinationStatusFactory::createForMonitorConfig(config('backup.monitor_backups'));
+
+        $this->expectException(DatabaseImportFailed::class);
+
+        $statuses->each(function (BackupDestinationStatus $backupDestinationStatus) {
+            $hd = new HealthySqlDump();
+            $hd->failsOnLackOfDiskSpace($backupDestinationStatus->backupDestination()->backups()->newest(), PHP_INT_MAX);
```

Comment:
A cleaner way to test this would be to have a dedicated class to check the disk space. Etc

```php
class DiskSpace {
    public function available() ....
    public function total() ....
}
```

In the failure method it should resolve by `app(DiskSpace::class)` and the in the test you can easily inject a mock by

```php
app()->bind(DiskSpace::class, new class {
    available() { return 0 };
    total() { return 123456 }; // doesn't matter
});
```

This way you can just run a normal backup:monitor and it should fail immediately